### PR TITLE
Decisions Module

### PR DESCRIPTION
**Pull request by kivSW**
The steps are coded but don't work yet. The tests for OneDrive work (except business account abilities )

### CODE REVIEW

**Comment by kivSW** [comment]
BuildModule.ps1 was added

File: Decisions.MSOneDrive/Steps/AbstractStep.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        public virtual OutcomeScenarioData[] OutcomeScenarios
+        {
+            get
+            {
+                return new OutcomeScenarioData[] { new OutcomeScenarioData(ERROR_OUTCOME, new DataDescription(typeof(OneDriveErrorInfo), ERROR_OUTCOME_DATA_NAME)) };
+            }
+        }
+
+        [EntityPickerEditor(new Type[] { typeof(OAuthToken) }, "MS OneDrive Token")]
+        //[TokenPicker]
```
Comment (ografov) [inline Decisions.MSOneDrive/Steps/AbstractStep.cs]: why not to use TokenPicker editor?

**Comment by kivSW** [comment]
The extra calling of CopyModule was removed

**Comment by kivSW** [comment]
the steps are ready

File: Decisions.MSOneDrive/Steps/AbstractStep.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        protected const string PERMISSION_ID = "Permission Id";
+        protected const string PARENT_FOLDER_ID = "Parent Folder Id";
+        protected const string LOCAL_FILE_PATH = "Local File Path";
+        protected const string PERMISSION = "Permission";
+        protected const string NEW_FOLDER_NAME = "New Folder Name";
+
+        protected const string TYPE_OF_LINK = "Type Of Link";
+        protected const string SCOPE_OF_LINK = "Scope of Link";
+
+        protected static T[] Concat<T>(T[] array, params T[] newItems)
```
Comment (ografov) [inline Decisions.MSOneDrive/Steps/AbstractStep.cs]: System.Linq.Enumerable.Concat

File: Decisions.MSOneDrive/Steps/AbstractStep.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        [WritableValue]
+        public string Token { get; set; }
+
+        private string FindAccessToken(string id)
+        {
+            ORM<OAuthToken> orm = new ORM<OAuthToken>();
+            var token = orm.Fetch(id);
+            if (token != null)
+                return token.TokenData;
+            throw new BusinessRuleException($"Can not find token with TokenId=\"{id}\"");
```
Comment (ografov) [inline Decisions.MSOneDrive/Steps/AbstractStep.cs]: EntityNotFoundException

**Review by ografov — APPROVED** [review]

**Comment by kivSW** [comment]
Code review corrections

**Pull request merged**